maintenance: update hardhat version to latest

antico5 opened the pull request:
Closes #305 

By updating the LSP's bundled hardhat to the latest version, we no longer get the related error.

kanej reviewed (commented):
Should we bump the version in the integration projects as well?

My mental model is that we have no specific reason for having a different hardhat version in any of the hardhat integration project samples right now, so the simplest thing is to keep them in-sync with the main project until we do have an example.

antico5 commented:
> 

I'll update the hardhat version of the test projects on the new protocol-tests branch, because I renamed the test/integration directory, so if I change it here there'll be a conflict when rebasing. Also just updated hardhat version to be 2.12.4

kanej reviewed (approved):
LGTM!
